src/hooks/bridge/index.ts

```diff
@@ -17,14 +17,22 @@ export const useEthBalanceOfAnotherChain = (chainId: ChainId | undefined) => {
   const { readProvider } = useKyberSwapConfig(chainId)
   const { account } = useActiveWeb3React()
   const [balance, setBalance] = useState<CurrencyAmount<Currency>>()
+  const controller = useRef(new AbortController())
+
   useEffect(() => {
+    controller.current.abort()
+    controller.current = new AbortController()
+    const signal = controller.current.signal
```

Comment:
why useRef insteadof cancel in clean up phase

Reply:
ah forgot this, updated